src/data/calendar.js

```diff
@@ -0,0 +1,136 @@
+    var numberOfWeeks = Math.ceil( used / 7);
+
+    var weeks = [];
+    for (var i = 0; i < numberOfWeeks; i++) {
+      weeks.push([{}, {}, {}, {}, {}, {}, {}]);
+    }
+    var firstWeekDayOfMonth = (firstOfMonth.getDay() - 1 + 7) % 7;
+
+    var frenchBankHolidays = getFrenchBankHolidays(year)
+    var bankHolidays = frenchBankHolidays
+        .map(holiday => year + "-" + zfill(holiday[0]) + "-" + zfill(holiday[1]));
+    var boxingDays = getBoxingDays(year, frenchBankHolidays)
+        .map(holiday => year + "-" + zfill(holiday[0]) + "-" + zfill(holiday[1]));
+
+    if (year < 2017) boxingDays = [];
+
+    return weeks.map((week, weekIndex) => {
+      return week.map((day, dayIndex) => {
+        var globalIndex = weekIndex * 7 + dayIndex;
+        var dayOfMonth = globalIndex - firstWeekDayOfMonth + 1;
+        day.date = new Date(year, month_number-1, dayOfMonth);
+
+        if (day.date.getMonth() === month_number - 1) {
+          var currentDate = year + "-" + zfill(day.date.getMonth() + 1) + "-" + zfill(dayOfMonth);
+          day.hours = 8;
+          if ([0, 6].includes(day.date.getDay())) {
+            day.hours = 8;
+            day.type = 'WE';
+          } else if (bankHolidays.includes(currentDate)) {
+            day.type = 'JF';
+          } else if (boxingDays.includes(currentDate)) {
+            day.type = 'CS';
+          } else {
+            day.type = 'JT';
+          }
+        } else {
+          day.type = null;
+        }
+        return day;
+      });
+    });
+  }
+
+  function updateWeeksWithHolidays(state) {
+    // Update the weeks data with the PTO infos for the current month.
+    state.holidays.forEach((holiday) => {
+      // Is in current month?
+      var start = new Date(holiday.start);
+      var end = new Date(holiday.end);
+
+      // Handle case where holidays ends on a week-end
+      while ([0, 6].includes(end.getDay())) {
+        end.setDate(end.getDate() - 1);
+      }
+
+      if (start.getMonth() <= state.currentMonth - 1 && start.getFullYear() <= state.currentYear &&
+          end.getMonth() >= state.currentMonth -1 && end.getFullYear() >= state.currentYear) {
+        // Someday of this holiday are in the current month
+        var hours = holiday.hours;
+        var type = guessTypeFromComment(holiday.comment);
+
+        state.weeks.forEach(week => {
+          week.forEach(day => {
+            var currentDayText = day.date.getFullYear() + '-' + zfill(day.date.getMonth() - 1) + '-' + zfill(day.date.getDate());
+            var startDayText = start.getFullYear() + '-' + zfill(start.getMonth() - 1) + '-' + zfill(start.getDate());
+            var endDayText = end.getFullYear() + '-' + zfill(end.getMonth() - 1) + '-' + zfill(end.getDate());
+
+            if (startDayText <= currentDayText && endDayText >= currentDayText) {
+              var isLast = endDayText == currentDayText;
+              if (day.type !== 'WE') {
+                if (day.type == 'JT') {
```

Comment:
at this point, type can be only WE, JT, CS or JF, is that so ? that's why I suggest a switch/case to make it clearer about what are the possible values at this point.